### PR TITLE
Migrate to yarn

### PR DESCRIPTION
**Pull request by TrySound**
One of NPM maintainers shows some disrespect to others which kills a
trust to them.

Yarn is more stable client with more full set of features so I suggest
to migrate to it.

As you can see yarn.lock is two times smaller because of less verbose
syntax. Lockfile is not required for every package in monorepo, thanks to
yarn workspaces.

### CODE REVIEW

**Comment by TrySound** [comment]
Original package-lock.json was gracefully converted to `yarn.lock` by its `yarn import` command.

**Comment by kof** [comment]
especially in the sight of getting rid of node_modules, I am now all in.

**Comment by kof** [comment]
Though if its really about npm inc, we should stop using registry as well haha

**Comment by kof** [comment]
What is the reason to actually use npm registry?

**Comment by TrySound** [comment]
Supporting own registry is quite expensive since all packages should be synchronised or lazy loaded from the same npm. So yarn registry for is is just an alias.

**Comment by kof** [comment]
No, I mean why not just use directly github?

**Comment by TrySound** [comment]
Because in this case you should push build artifacts which is verbose for history. There is also a problem with loading all this stuff. Repos may be quite big. And monorepo's won't work with github links.

**Comment by kof** [comment]
true

**Comment by kof** [comment]
I will talk to https://unpkg.com/ guys to see what we can do there.

**Comment by kof** [comment]
We just need to be able to publish to a cdn like unpkg without npm registry and thats it.

**Comment by kof** [comment]
can you please also update docs where npm was used? just remove npm completely

File: scripts/build.js [hunk 1]
```diff
@@ -25,7 +24,7 @@ function build(options = {}) {
 
   if (!fs.existsSync('src')) {
     if (!silent) {
-      log.error('No src dir')
+      console.error('No src dir')
```
Comment (kof) [inline scripts/build.js]: linter warnings, need to add eslint-disable statement

**Comment by kof** [comment]
Merged!!!! 🎉🎉🎉🎉

**Comment by HenriBeck** [comment]
I think yarn is better client though not publishing to npm and only to unpkg would be a mistake in my opinion. I think we would lose a lot of users through that.

**Comment by kof** [comment]
We can publish, but we could avoid using it ourselves.

**Comment by kof** [comment]
Btw. does yarn has its own registry or is it just using npm registry?

**Comment by HenriBeck** [comment]
The yarn registry being a mere mirror of the npm registry. Though this is being removed in yarn v2 I think.

**Comment by TrySound** [comment]
I think the best is having central registry with packages and if as a fallback distributed database like torrent.

**Comment by TrySound** [comment]
Yarn registry is already removed I think. Currently it's just a proxy.

**Pull request merged**